joomla/joomla-cms
# Fixes to Add fast check for image orientation (#3568)

#### Description

This fixes some issues from #3568:
- it get rids of the new (**yet-to-be-released!**) JImageHelper class by incorporating its code into the JImage class.
- it correct some comment blocks
#### Test instruction

Assuming you are using the Protostar template, modify the template's index.php by inserting the following code after line 27 (`$sitename = $app->get('sitename');`):

```
echo '<pre>';

// Define the file to use. 
// NOTE:   Make sure this file exists or use another path to an existing image file!
$file = JPATH_BASE . '/images/powered_by.png';

echo str_repeat('_', 60) . "\n\n";
echo 'File is: ' . print_r($file, true) . "\n";
echo str_repeat('_', 60) . "\n\n";

// Get instance of JImage.
$image = new JImage($file); // $image must be instance of JImage

$orientation = $image->getOrientation();
echo 'For the $image object orientation is: ' . print_r($orientation, true) . "\n";
echo(str_repeat('_', 60)) . "\n\n";

// Get image properties calling static method from JImage::getImageFileProperties().
$props = JImage::getImageFileProperties($file);

echo 'Properties from JImage::getImageFileProperties($file);:';
echo "\n\n";
print_r($props);
echo "\n";
echo (str_repeat('_', 60)) . "\n\n";

echo 'Check for every property separately:' . "\n\n";

// Check the returned data contains all of these properties.
foreach (array(
   'width',
   'height',
   'type',
   'attributes',
   'bits',
   'channels',
   'mime',
   'filesize',
   'orientation') as $p
)
{
    if (array_key_exists($p, $props))
    {
        echo "{$p} is: " . print_r($props->$p, true) . "\n";
    }
    else
    {
        echo print_r("\$props does not contain {$p}-data", true) . "\n";
    }
}

echo(str_repeat('_', 60)) . "\n";
echo '</pre>';
die;
```

Visit your test site. The expected output is:

```
____________________________________________________________

File is: _PATH_TO_YOUR_JPATH_BASE_HERE_/images/powered_by.png
____________________________________________________________

For the $image object orientation is: landscape
____________________________________________________________

Properties from JImage::getImageFileProperties($file);:

stdClass Object
(
    [width] => 150
    [height] => 45
    [type] => 3
    [attributes] => width="150" height="45"
    [bits] => 8
    [channels] => 
    [mime] => image/png
    [filesize] => 4048
    [orientation] => landscape
)
____________________________________________________________

Check for every property separately:

width is: 150
height is: 45
type is: 3
attributes is: width="150" height="45"
bits is: 8
channels is: 
mime is: image/png
filesize is: 4048
orientation is: landscape
____________________________________________________________
```
